pyne/vta.py

```diff
@@ -0,0 +1,566 @@
+"""Purpose:
+
+This module provides a modified ray traverse method, which is modifed based on
+http://www.cse.yorku.ca/~amana/research/grid.pdf, to calculate which voxels
+intersect with a given ray (line segment).
+Several rays are used to calculate the voxels that intersect with the facet.
+A series of rays are constructed based on both the facet and the mesh grid to
+make sure all the wanted voxels will be found. Combining all the found voxels
+for each ray, we get the voxels intersecting with the facet.
+
+These voxels will be used to construct the bounding volume for source sampling.
+But currently these functions are separate from MOAB. Interface of MOAB with
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+class Point(object):
```

Comment:
Make Point iterable.

Reply:
Maybe just a numpy array of size 3.

Reply:
One option for this is to make a list of the x,y,z values and index that list. This would allow you to keep the nicely named attributes if you're referring to a single value, but still append to lists or combine points cleanly. 

```
def __iter__(self, key):
    l = list(self.x, self.y, self.z)
    return l[key]
```